samza-kafka/src/test/java/org/apache/samza/system/kafka/TestKafkaConsumerProxy.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.system.kafka;
+
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.junit.Test;
+
+import static org.mockito.Mockito.anyCollection;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+
+
+public class TestKafkaConsumerProxy {
+
+  @Test
+  public void testSynchronizedAccessOfKafkaConsumer() throws Exception {
```

Comment:
I see we don't have tests at all for consumer proxy and thanks for adding one :) However, I have few observations/questions
1. The test still does have some indeterminism and doesn't necessarily tests synchronized access as it claims all the time as it is possible that t1 & t2 run exclusively.
2. Is the intent here to test java `synchronized` or prevent accidental changes to initializeLags making it no longer synchronize on the consumer? 
- If it is the former, it is going to be hard to test consistently and deterministically unless you do something complex gimmicks with latches & `Thread.holdsLock(...)`. IMO, it should be okay to assume Java `synchronized` works as expected
- If it the latter, it is still undoubtedly hard but we can enforce them through other means with useful explicit comments around the code blocks & diligent reviews. 

TLDR; I am fine with not having a test for this scenario instead of having one that is indeterministic in what it is tests. If you prefer to have a test, then it will be nice to have it do what it claims to do.

Reply:
haha, i debated for a good deal of time about this test and even having a test for this scenario :D 
I agree that this test can not really "test/simulate" the real issue we are solving. I wonder if there can be a test which can really test this scenario. I will remove this test.